java/sage/media/bluray/BluRayFile.java

```diff
@@ -0,0 +1,496 @@
+  {
+    // Check to see if we need to move to a different file
+    if (forceOpen || pos < fileOffsets[currFileIndex] || (currFileIndex < fileOffsets.length - 1 && pos >= fileOffsets[currFileIndex + 1]))
+    {
+      int oldIndex = currFileIndex;
+      for (currFileIndex = 0; currFileIndex < fileOffsets.length; currFileIndex++)
+      {
+        if (fileOffsets[currFileIndex] > pos)
+          break;
+      }
+      currFileIndex--;
+      if (sage.Sage.DBG) System.out.println("Switching BluRay source file from index " + oldIndex + " to " + currFileIndex);
+      int currAngle = Math.min(prefAngle, currPlaylist.playlistItems[currFileIndex].itemClips.length - 1);
+      sageFileChannel.openFile(
+          new java.io.File(streamDir,
+              currPlaylist.playlistItems[currFileIndex].itemClips[currAngle].clipName +
+                  (bdp.doesUseShortFilenames() ? ".MTS" : ".m2ts")).getAbsolutePath());
+    }
+  }
+
+  private long realPosition()
+  {
+    return position() - fileOffsets[currFileIndex];
+  }
+
+  @Override
+  public long getBytesLeftInClip()
+  {
+    return (currFileIndex < fileOffsets.length - 1) ? (fileOffsets[currFileIndex + 1] - position()) : (totalSize - position());
+  }
+
+  @Override
+  public int getCurrClipIndex()
+  {
+    return currFileIndex;
+  }
+
+  @Override
+  public long getClipPtsOffset(int index)
+  {
+    return ptsOffsets[index];
+  }
+
+  @Override
+  public int getClipIndexForNextRead()
+  {
+    if (getBytesLeftInClip() == 0 && currFileIndex < fileOffsets.length - 1)
+      return currFileIndex + 1;
+    else
+      return currFileIndex;
+  }
+
+  @Override
+  public ContainerFormat getFileFormat()
+  {
+    return bdp.getFileFormat(targetTitle - 1);
+  }
+
+  @Override
+  public int getTitle()
+  {
+    return targetTitle;
+  }
+
+  @Override
+  public long getChapterStartMsec(int chapter)
+  {
+    return chapterOffsets[Math.max(0, Math.min(chapter - 1, chapterOffsets.length - 1))] / 45;
+  }
+
+  @Override
+  public int getNumTitles()
+  {
+    return bdp.getNumPlaylists();
+  }
+
+  @Override
+  public int getNumChapters()
+  {
+    return chapterOffsets.length;
+  }
+
+  @Override
+  public int getChapter(long pts45)
+  {
+    for (int i = 0; i < chapterOffsets.length; i++)
+      if (chapterOffsets[i] > pts45)
+        return i;
+    return chapterOffsets.length;
+  }
+
+  @Override
+  public String getTitleDesc(int titleNum)
+  {
+    return bdp.getPlaylistDesc(titleNum - 1);
+  }
+
+  @Override
+  public int getNumAngles()
+  {
+    return currPlaylist.playlistItems[currFileIndex].itemClips.length;
+  }
+
+  @Override
+  public long transferTo(long position, long count, WritableByteChannel target) throws IOException
+  {
+    long bytesRead = 0;
+
+    try
+    {
+      ensureProperFile(false, position);
+      bytesRead = sageFileChannel.transferTo(position - fileOffsets[currFileIndex], count, target);
+
+      if (bytesRead < count)
```

Comment:
You can just make this a 'while' instead of an 'if' and then remove the above 2 lines of code...then it will also cover spanning N number of files rather than just 2 (although, I don't think it'll ever be more than 2 in reality).